### PR TITLE
Add setters & getters for various MCF sections; rename package

### LINKED ISSUE #7: switch to Apache 2 license to match our other software

### PR DESCRIPTION
**Pull request by davemfish**
This PR adds methods for populating most of the metadata sections described in #6 , to the extent that they are supported by the MCF schema.

Also in the PR;
* renamed the package
* added apidocs builds via sphinx
* switched the license to Apache 2. Fix #7

### CODE REVIEW

File: tests/test_geometamaker.py
Comment (davemfish) [inline tests/test_geometamaker.py:1]: Sorry about the lack of a diff here. git knows I renamed the file, but github seems to think it was an "add + delete" situation.

**Review by empavia — APPROVED** [review]
This seems like it incorporates the items we discussed! I don't see anything glaring to comment on. I am not as familiar with the logistics of making a library.

File: src/geometamaker/geometamaker.py [hunk 1]
```diff
@@ -271,9 +331,85 @@ def set_keywords(self, keywords, section='default', keywords_type='theme',
...
+        license_dict['url'] = license_url if license_url else ''
+        self.mcf['identification']['license'] = license_dict
+        self.mcf['identification']['accessconstraints'] = constraints
+        self.validate()
+
+    def get_license(self):
+        """Get `license` for the dataset.
+
+        Returns:
+            dict or `None` if `license` does not exist.
```
Comment (phargogh) [inline src/geometamaker/geometamaker.py:363]:
```suggestion
            dict or ``None`` if ``license`` does not exist.
```

This is pretty minor, but isn't sphinx configured here to interpret this as RST?  If so, we'd need double-backticks to have it interpreted as inline code.

**Review by phargogh — COMMENTED** [review]
This is looking good Dave!  My only comments here are kind of general, specifically:

* There are a couple places where RST double-backticks should be used to represent inline code (assuming the goal is to use RST syntax in docstrings) instead of the MD-style single-backticks
* Running `pytest-cov` I see that there are a couple of edge cases that aren't yet tested.  I'm not sure that should hold up this PR, so I just wanted to mention it.

Otherwise, it's looking good!

**Comment by davemfish** [comment]
> This is looking good Dave! My only comments here are kind of general, specifically:
> 
> * There are a couple places where RST double-backticks should be used to represent inline code (assuming the goal is to use RST syntax in docstrings) instead of the MD-style single-backticks
> * Running `pytest-cov` I see that there are a couple of edge cases that aren't yet tested.  I'm not sure that should hold up this PR, so I just wanted to mention it.
> 
> Otherwise, it's looking good!

Thanks, @phargogh . I pushed a commit fixing the backticks. And thanks for prompting me to check the test coverage. It looks like all the missing coverage are in areas that weren't otherwise touched in this PR. So I'm just going to make a separate issue for it.

**Review by phargogh — APPROVED** [review]

**Pull request merged**